tests/functional/mesh/reload_test.go

```diff
@@ -0,0 +1,266 @@
+		testGroup := subtest.testGroup
+		shortCommand := subtest.shortCommand
+		longCommand := subtest.longCommand
+		extraLongCommand := subtest.extraLongCommand
+		workSetup := func(testName string) (map[string]*receptorcontrol.ReceptorControl, *mesh.CLIMesh, []byte) {
+			data := mesh.YamlData{}
+			data.Nodes = make(map[string]*mesh.YamlNode)
+			expectedResults := []byte("1\n2\n3\n4\n5\n")
+			// Generate a mesh with 3 nodes
+
+			data.Nodes["node2"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef: []interface{}{
+					map[interface{}]interface{}{
+						"tcp-listener": map[interface{}]interface{}{
+							"cost": 1.0,
+							"nodecost": map[interface{}]interface{}{
+								"node1": 1.0,
+								"node3": 1.0,
+							},
+						},
+					},
+					shortCommand,
+				},
+			}
+			data.Nodes["node1"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{
+					"node2": {
+						Index: 0,
+					},
+				},
+				Nodedef: []interface{}{},
+			}
+			data.Nodes["node3"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{
+					"node2": {
+						Index: 0,
+					},
+				},
+				Nodedef: []interface{}{
+					longCommand,
+					shortCommand,
+					extraLongCommand,
+				},
+			}
+
+			m, err := mesh.NewCLIMeshFromYaml(data, testName)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ctx, _ := context.WithTimeout(context.Background(), 60*time.Second)
+			err = m.WaitForReady(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			nodes := m.Nodes()
+			controllers := make(map[string]*receptorcontrol.ReceptorControl)
+			for k := range nodes {
+				controller := receptorcontrol.New()
+				err = controller.Connect(nodes[k].ControlSocket())
+				if err != nil {
+					t.Fatal(err)
+				}
+				controllers[k] = controller
+			}
+
+			return controllers, m, expectedResults
+		}
+
+		tearDown := func(controllers map[string]*receptorcontrol.ReceptorControl, m *mesh.CLIMesh) {
+			defer m.WaitForShutdown()
+			defer m.Destroy()
+			defer func() {
+				for _, controller := range controllers {
+					controller.Close()
+				}
+			}()
+		}
+
+		t.Run(testGroup+"/Reload mesh ", func(t *testing.T) {
+			t.Parallel()
+			controllers, m, _ := workSetup(t.Name())
+			defer tearDown(controllers, m)
+
+			// submit work from node 2 to node 3
+			unitID, err := controllers["node2"].WorkSubmit("node3", "echosleeplong50")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// wait for 10 seconds, and check if the work is in pending state
+			ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+			err = controllers["node2"].AssertWorkPending(ctx, unitID)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// declare a new mesh with no connection between nodes
+			modifiedData := mesh.YamlData{}
+			modifiedData.Nodes = make(map[string]*mesh.YamlNode)
+			modifiedData.Nodes["node2"] = &mesh.YamlNode{
+				Connections: map[string]mesh.YamlConnection{},
+				Nodedef: []interface{}{
```

Comment:
I don't think setting nodedef is necessary here

Reply:
I have made the changes